source/fundamentals/builders/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1198 @@
+                        var totalSeats = showtime
+                              .<MqlInteger>getArray("seats")
+                              .sum(n -> n);
+                        var isAvailable = showtime
+                              .getInteger("ticketsBought")
+                              .lt(totalSeats);
+                        return isAvailable;
+                  })
+   ))));
+
+.. note::
+
+   To improve readability, the previous example assigns intermediary values to
+   the ``totalSeats`` and ``isAvailable`` variables. If you don't pull
+   out these intermediary values as variables, the code still produces
+   equivalent results.
+
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [
+     {
+       $project: {
+         availableShowtimes: {
+           $filter: {
+             input: "$showtimes",
+             as: "showtime",
+             cond: { $lt: [ "$$showtime.ticketsBought", { $sum: "$$showtime.seats" } ] }
+   }}}}]
+
+Boolean Operations
+~~~~~~~~~~~~~~~~~~
+
+You can perform a boolean operation on a value of type ``MqlBoolean``
+using the methods described in this section.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Java Method
+     - Aggregation Pipeline Operator
+
+   * - `and() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#and(com.mongodb.client.model.mql.MqlBoolean)>`__
+     - :manual:`$and </reference/operator/aggregation/and/>`
+       
+   * - `not() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#not()>`__
+     - :manual:`$not </reference/operator/aggregation/not/>`
+
+   * - `or() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#or(com.mongodb.client.model.mql.MqlBoolean)>`__
+     - :manual:`$or </reference/operator/aggregation/or/>`
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+Suppose you want to classify very low or high weather temperature
+readings (in degrees Fahrenheit) as extreme.
+
+The ``or()`` operator checks to see if temperatures are extreme by comparing
+the ``temperature`` field to predefined values with ``lt()`` and ``gt()``.
+The ``project()`` method records this result in the ``extremeTemp`` field.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   var temperature = current().getInteger("temperature");
+
+   asList(project(fields(
+            computed("extremeTemp",
+               temperature
+                  .lt(of(10))
+                  .or(temperature.gt(of(95)))
+   ))));
+
+The following code provides an equivalent aggregation pipeline in
+the Query API:
+
+.. code-block:: javascript
+   :copyable: true
+
+   [
+     {
+       $project: {
+         extremeTemp: { $or: [ { $lt: ["$temperature", 10] },
+                               { $gt: ["$temperature", 95] } ] }
+   }}]
```

Comment:
```suggestion
   [ { $project: {
       extremeTemp: { $or: [ { $lt: ["$temperature", 10] },
                             { $gt: ["$temperature", 95] } ] }
   }}]
```
I'm not sure what the spacing should be. Some of the }]}}]s are spaced, but others are not. But if we're going to condense, I think we should condense the first two.

Reply:
Yes, these are tricky.

Reply:
also making these 4-spaced